File: client/packages/post_repository/lib/src/models/post.dart
import 'package:equatable/equatable.dart';

class Post extends Equatable {
  const Post({
    required this.id,
    required this.content,
    required this.tags,
    required this.createdAt,
    required this.anonymous,
    this.givenReactionId,
    this.reactions,
  });

  Post.fromMap(Map<String, dynamic> map)
      : id = map['id'] as String,
        content = map['content'] as String,
        tags = List<int>.from(map['tags'] as List),
        createdAt = DateTime.parse(map['createdAt'] as String),
        anonymous = map['anonymous'] as bool,
        givenReactionId = map['givenReactionId'] as int?,
        reactions = Map<String, int?>.from(map['reactions'] as Map);

  final String id;
  final String content;
  final List<int> tags;
  final DateTime createdAt;
  final bool anonymous;
  final int? givenReactionId;
  final Map<String, int?>? reactions;

  @override
  List<Object?> get props => [
        id,
        content,
        tags,
        createdAt,
        anonymous,
        givenReactionId,
        reactions,
      ];
}

File: backend/functions/src/model/post.ts
export class Post {
  id: string
  content: string
  tags: number[]
  createdAt: String
  anonymous: boolean
  givenReactionId: number | undefined
  reactions: [string, unknown][] | undefined

  constructor(id: string, content: string, tags: number[], createdAt: Date, anonymous: boolean,
              givenReactionId: number | undefined, reactions: [string, unknown][] | undefined) {
    this.id = id;
    this.content = content;
    this.tags = tags;
    this.createdAt = createdAt.toJSON();
    this.anonymous = anonymous;
    this.givenReactionId = givenReactionId;
    this.reactions = reactions;
  }
}
